base/strings/util.jl

```diff
@@ -464,11 +464,87 @@ function hex2bytes(s::AbstractString)
 end
 
 """
+    hex2bytes(s::AbstractVector{UInt8})
+
+Convert the hexadecimal bytes array to its binary representation. Returns
+`Vector{UInt8}`, i.e. a vector of bytes.
+"""
+@inline function hex2bytes(s::AbstractVector{UInt8})
+    d = zeros(UInt8, div(endof(s), 2))
+    return hex2bytes!(d, s)
+end
+
+"""
+    hex2bytes!(d::AbstractVector{UInt8}, s::AbstractVector{UInt8})
+
+Convert the hexadecimal bytes vector to its binary representation. The results are
+populated into a destination vector. The function returns the destination vector.
+
+# Examples
+```jldoctest
+julia> s = UInt8["01abEF"...]
+6-element Array{UInt8,1}:
+ 0x30
+ 0x31
+ 0x61
+ 0x62
+ 0x45
+ 0x46
+
+julia> d =zeros(UInt8, 3)
+3-element Array{UInt8,1}:
+ 0x00
+ 0x00
+ 0x00
+
+julia> hex2bytes!(d, s)
+3-element Array{UInt8,1}:
+ 0x01
+ 0xab
+ 0xef
+```
+"""
+function hex2bytes!(d::AbstractVector{UInt8}, s::AbstractVector{UInt8})
+    i, j = start(s), 0
+    # This line is important as this ensures computation happens in word boundary and not
+    # byte boundary. Boundary computation can be almost 10 times slower
+    n::UInt = 0
+    c1::UInt = 0
+    c2::UInt = 0
+    while !done(s, i)
+        n = 0
+        c1, i = next(s, i)
+        done(s, i) && throw(ArgumentError(
+            "string length must be even: length($(repr(s))) == $(length(s))"))
+        c2, i = next(s, i)
+        n = number_from_hex(c1)
+        n <<= 4
+        n += number_from_hex(c2)
+        d[j+=1] = (n & 0xFF)
+    end
+    resize!(d, j)
```

Comment:
dependency on length removed. The array should not be printed as it an be large. 

-